### PR TITLE
Fix a couple of formatter issues

### LINKED ISSUE #7599: Bad comment formatting on next line when there's a comment after def
```cr
def method # this is a method
  # this is a comment
end
```
gets formatted to:
```cr
def method # this is a method
# this is a comment
end
```
But I expect nothing to happen.

### LINKED ISSUE #7600: Bad formatting for comma enum with comments
```cr
enum E
  A, # hello
  B, # hello
  C, # hello
end
```
gets formatted to:
```cr
enum E
  A,  # hello
B,    # hello
C,    # hello
end
```
But I expect nothing to happen.

### PR DESCRIPTION
**Pull request by asterite**
Fixes #7599
Fixes #7600

### CODE REVIEW

File: src/compiler/crystal/tools/formatter.cr [hunk 1]
```diff
@@ -2058,9 +2061,16 @@ module Crystal
 
       # This is the case of an enum member
       if node.name[0].ascii_uppercase? && @token.type == :","
```
Comment (j8r) [inline src/compiler/crystal/tools/formatter.cr]: Is there any advantage of supporting both the comma and space as member delimiters in enums?

Comment (asterite) [reply src/compiler/crystal/tools/formatter.cr]: That space is supported is a bug. Only comma or newline should be supported. Please open a bug for this.

Comment (j8r) [reply src/compiler/crystal/tools/formatter.cr]: Excluding the comma, why supporting both the comma and new line (and therefore the semicolon)?
```cr
enum E
  A; B; C
  D, E, F
end
```

Comment (asterite) [reply src/compiler/crystal/tools/formatter.cr]: I copied the comma from C#. I wouldn't mind dropping it, but it's a breaking change (I don't mind either)

**Comment by bcardiff** [comment]
Dropping the comma from the formatter and from the parser can be done in another PR

**Pull request merged**